doubleunion/arooo
# Update Rails to 5.2.4.3

### What does this code do, and why?

This change upgrades our version of Rails to be 5.2.4.3, a whole major version ahead of what we have. It takes us off of a no longer maintained version of Rails, removes some deprecation warnings, and makes us better compatible with the version of Ruby we are already using.

This PR addresses issues: https://github.com/doubleunion/arooo/issues/459 and https://github.com/doubleunion/arooo/issues/457

However, this upgrade does not completely remove all deprecation warnings. In particular, we still have a couple of Ruby-level warnings:
* Many `warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call`, coming from various Rails libraries that we can't easily upgrade further (activesupport, activemodel, etc). These are due to older Rails code being used with Ruby 2.7.1. The Rails community is fixing these for Rails 6.0 (see e.g. https://github.com/rails/rails/pull/37935), but it is unclear if these will be ported to Rails 5.*.
  * At least one of these warnings is coming from `stripe-ruby-mock`, which might be possible to upgrade, but would require careful testing. It might be worth opening a new issue and trying that in a follow-up PR.
* Some `warning: rb_check_safe_obj will be removed in Ruby 3.0`, also coming from `activesupport`.

It is possible to suppress the deprecation warnings across the board, for all of the code, but that seems risky. I could not find a way to suppress only warnings for a specific package, so we'll have to live with those warnings for now.

One thing I see in 


### How is this code tested?

* Specs pass on CI
* Manually tested on local server:
  * logging in with GitHub
  * saving an application without submitting it
  * submitting an application
  * made applicant a member via Rails console
  * became key member via the app
  * added Google Auth
  * verified that app looks fine (no obviously missing styles or assets) for logged-in member
  * verified that member can't see "admin" stuff in app
  * verified that, with admin bit set to true, one can see the admin area, with the expected sections

### Are any database migrations required by this change?

Nope.

### Screenshots (before/after)

Not relevant.

### Are there any configuration or environment changes needed?

No.
